ingestion-beam/src/main/java/com/mozilla/telemetry/transforms/PubsubMessageToTableRow.java

```diff
@@ -323,13 +344,19 @@ private void processField(String jsonFieldName, Field field, Object val,
     } else if (field.getType() == LegacySQLTypeName.RECORD && field.getMode() == Mode.REPEATED) {
       List<Object> records = value.filter(List.class::isInstance).map(List.class::cast)
           .orElse(ImmutableList.of());
+      List<Object> repeatedAdditionalProperties = new ArrayList<>();
       records.stream().filter(Map.class::isInstance).map(Map.class::cast).forEach(record -> {
         Map<String, Object> props = additionalProperties == null ? null : new HashMap<>();
         transformForBqSchema(record, field.getSubFields(), props);
         if (props != null && !props.isEmpty()) {
-          additionalProperties.put(name, props);
+          repeatedAdditionalProperties.add(props);
+        } else {
+          repeatedAdditionalProperties.add(null);
         }
       });
+      if (!repeatedAdditionalProperties.stream().allMatch(p -> p == null)) {
```

Comment:
```suggestion
      if (!repeatedAdditionalProperties.stream().allMatch(Objects::isNull)) {
```

Slight preference for a method reference here.

Reply:
And this approach to padding with nulls to preserve index of additional properties is very nice.